core/src/main/kotlin/com/strumenta/kolasu/model/Naming.kt

```diff
@@ -65,3 +67,27 @@ fun <N> ReferenceByName<N>.tryToResolve(possibleValue: N?): Boolean where N : Po
         true
     }
 }
+
+interface Symbol : PossiblyNamed
+
+data class Scope(
+    val symbols: MutableMap<String, MutableList<Symbol>> = mutableMapOf(),
+    val parent: Scope? = null,
+) {
+    constructor(symbols: MutableList<Symbol>, parent: Scope? = null) : this(
```

Comment:
here the list could also be immutable. I think List<Symbol> is an ancestor of MutableList

Reply:
:+1: